framework/src/transfers/MultiAppMeshFunctionTransfer.C

```diff
@@ -94,350 +100,354 @@ MultiAppMeshFunctionTransfer::execute()
+        MeshBase::const_element_iterator elem_end = to_mesh->local_elements_end();
 
-        unsigned int lowest_app_rank = libMesh::invalid_uint;
-        Real best_val = 0;
-        bool point_found = false;
-        for (unsigned int i_proc = 0; i_proc < incoming_evals.size(); i_proc++)
+        for (; elem_it != elem_end; ++elem_it)
         {
-          // Skip this proc if the elem wasn't in it's bounding boxes.
-          std::pair<unsigned int, unsigned int> key(i_to, elem->id());
-          if (point_index_map[i_proc].find(key) == point_index_map[i_proc].end())
+          Elem * elem = *elem_it;
+
+          // Skip this element if the variable has no dofs at it.
+          if (elem->n_dofs(sys_num, var_num) < 1)
             continue;
-          unsigned int i_pt = point_index_map[i_proc][key];
 
-          // Ignore this proc if it's app has a higher rank than the
-          // previously found lowest app rank.
-          if (_direction == FROM_MULTIAPP)
+          unsigned int lowest_app_rank = libMesh::invalid_uint;
+          Real best_val = 0;
+          bool point_found = false;
+          for (unsigned int i_proc = 0; i_proc < incoming_evals.size(); i_proc++)
           {
-            if (incoming_app_ids[i_proc][i_pt] >= lowest_app_rank)
+            // Skip this proc if the elem wasn't in it's bounding boxes.
+            std::pair<unsigned int, unsigned int> key(i_to, elem->id());
+            if (point_index_map[i_proc].find(key) == point_index_map[i_proc].end())
               continue;
-          }
+            unsigned int i_pt = point_index_map[i_proc][key];
 
-          // Ignore this proc if the point was actually outside its meshes.
-          if (incoming_evals[i_proc][i_pt] == OutOfMeshValue)
-            continue;
+            // Ignore this proc if it's app has a higher rank than the
+            // previously found lowest app rank.
+            if (_direction == FROM_MULTIAPP)
+            {
+              if (incoming_app_ids[i_proc][i_pt] >= lowest_app_rank)
+                continue;
+            }
 
-          best_val = incoming_evals[i_proc][i_pt];
-          point_found = true;
-        }
+            // Ignore this proc if the point was actually outside its meshes.
+            if (incoming_evals[i_proc][i_pt] == OutOfMeshValue)
+              continue;
+
+            best_val = incoming_evals[i_proc][i_pt];
+            point_found = true;
+          }
 
-        if (_error_on_miss && !point_found)
-          mooseError("Point not found! ", elem->centroid() + _to_positions[i_to]);
+          if (_error_on_miss && !point_found)
+            mooseError("Point not found! ", elem->centroid() + _to_positions[i_to]);
 
-        dof_id_type dof = elem->dof_number(sys_num, var_num, 0);
-        solution->set(dof, best_val);
+          dof_id_type dof = elem->dof_number(sys_num, var_num, 0);
+          solution->set(dof, best_val);
+        }
       }
+      solution->close();
+      to_sys->update();
```

Comment:
These statements can remain in the outer loop. We don't need to close and update the solution for every variable.

Reply:
I'm fine with that. But ```solution``` is defined within the loop so I would have to move its definition before the outer loop (and make it either a member variable or a function argument if I am to move that code into a subroutine)

Is that what you have in mind?

Reply:
Right, either is fine with me. I assume the subroutine will be private to the object so it doesn't matter to me if you go with member or argument as long as it's coded correctly.

Reply:
Actually, these two lines need to be inside the ```for (unsigned int i_to = 0; i_to < _to_problems.size(); i_to++)``` loop because ```to_sys``` depends on ```i_to```